src/backend/Level0/CHIPBackendLevel0.cc

```diff
@@ -694,72 +693,108 @@ CHIPEventLevel0 *CHIPQueueLevel0::getLastEvent() {
   return (CHIPEventLevel0 *)LastEvent_;
 }
 
-ze_command_list_handle_t CHIPQueueLevel0::getCmdListCopy() {
-#ifdef L0_IMM_QUEUES
-  return ZeCmdListCopyImm_;
-#else
-  ze_command_list_handle_t CommandList;
-  std::lock_guard<std::mutex> LockQueues(Backend->QueueCreateDestroyMtx);
-  auto Status = zeCommandListCreate(ZeCtx_, ZeDev_, &CommandListMemoryDesc_,
-                                    &CommandList);
-  CHIPERR_CHECK_LOG_AND_THROW(Status, ZE_RESULT_SUCCESS,
-                              hipErrorInitializationError);
-  return CommandList;
-#endif
-}
-
-ze_command_list_handle_t CHIPQueueLevel0::getCmdListCompute() {
+ze_command_list_handle_t CHIPQueueLevel0::getCmdList() {
 #ifdef L0_IMM_QUEUES
-  return ZeCmdListComputeImm_;
+  return ZeCmdList_;
 #else
   ze_command_list_handle_t ZeCmdList;
   std::lock_guard<std::mutex> LockQueues(Backend->QueueCreateDestroyMtx);
   auto Status =
-      zeCommandListCreate(ZeCtx_, ZeDev_, &CommandListComputeDesc_, &ZeCmdList);
+      zeCommandListCreate(ZeCtx_, ZeDev_, &CommandListDesc_, &ZeCmdList);
   CHIPERR_CHECK_LOG_AND_THROW(Status, ZE_RESULT_SUCCESS,
                               hipErrorInitializationError);
   return ZeCmdList;
 #endif
 }
 
+CHIPQueueLevel0::CHIPQueueLevel0(CHIPDeviceLevel0 *ChipDev)
+    : CHIPQueueLevel0(ChipDev, 0, L0_DEFAULT_QUEUE_PRIORITY,
+                      LevelZeroQueueType::Compute) {}
+
 CHIPQueueLevel0::CHIPQueueLevel0(CHIPDeviceLevel0 *ChipDev,
                                  CHIPQueueFlags Flags)
-    : CHIPQueueLevel0(ChipDev, Flags, L0_DEFAULT_QUEUE_PRIORITY) {}
-CHIPQueueLevel0::CHIPQueueLevel0(CHIPDeviceLevel0 *ChipDev)
-    : CHIPQueueLevel0(ChipDev, 0, L0_DEFAULT_QUEUE_PRIORITY) {}
+    : CHIPQueueLevel0(ChipDev, Flags, L0_DEFAULT_QUEUE_PRIORITY,
+                      LevelZeroQueueType::Compute) {}
+
+CHIPQueueLevel0::CHIPQueueLevel0(CHIPDeviceLevel0 *ChipDev,
+                                 CHIPQueueFlags Flags, int Priority)
+    : CHIPQueueLevel0(ChipDev, Flags, Priority, LevelZeroQueueType::Compute) {}
+
+CHIPQueueLevel0::CHIPQueueLevel0(CHIPDeviceLevel0 *ChipDev,
+                                 CHIPQueueFlags Flags, int Priority,
+                                 LevelZeroQueueType TheType)
```

Comment:
What is the use of TheType? I thought CHIP_SPV maintains compute and copy descriptors(ordinal, index and other details) per queue so that based on workload it can choose one of them. Did we change it now?
If TheType is passed at the the time of construction then queue will be initialized with one type and moving to other may not be possible. Is my understanding correct?

Reply:
This PR ended up being much bigger than I expected currently the copy queues, if available, will not be used. I will implement this optimization in a subsequent PR.  @Sarbojit2019

Reply:
> What is the use of TheType? I thought CHIP_SPV maintains compute and copy descriptors(ordinal, index and other details) per queue so that based on workload it can choose one of them. Did we change it now?

Yes we did. The descriptors for copy and compute live in the `CHIPDeviceLevel0` now. A Level Zero queue can now be either of compute type or copy type. Having multiple queues inside a single `CHIPQueue` will get confusing when having to sync queues, etc.